### PR TITLE
initiating frontend dev

### PR DESCRIPTION
**Pull request by tjingrant**
There are several questions I'm currently still searching for answers to..
1. Best way to specify inputs/outputs. 
- Currently I take Placeholder node as an input.
- Currently I require the proto definition of output node to be passed in. 
2. Shape of output is not present in TF GraphDef.

I'll look into it more on Monday.

### CODE REVIEW

File: onnx_tf/frontend.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  # DT_HALF = 19;
+  # DT_RESOURCE = 20;
+  # DT_VARIANT = 21;  // Arbitrary C++ data types
+  # DT_UINT32 = 22;
+  # DT_UINT64 = 23;
+  ]
+
+  @classmethod
+  def op_name_to_lower(cls, name):
+    return re.sub('(?<!^)(?=[A-Z])', '_', name).lower()
```
Comment (arpith-jacob) [inline onnx_tf/frontend.py]: Is this also sanitizing names?  Could you add a comment on the rules?

File: test/frontend/test_node.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    for node in tf_graph.node:
+        if node.name == "Relu":
+            output_node = node
+
+    onnx_graph = convert_graph(tf_graph, output_node)
+    tf_rep = prepare(helper.make_model(onnx_graph))
+    in_tensor = self._get_rnd(shape)
+    output = tf_rep.run({"Placeholder": in_tensor})["Relu"]
+
+    np.testing.assert_allclose(output, np.clip(in_tensor, 0, 1.0))
```
Comment (arpith-jacob) [inline test/frontend/test_node.py]: Is it possible to compare the result of evaluating 'tf_graph' to 'output'?

File: onnx_tf/frontend.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      else:
+        handler_name = "handle_" + cls.op_name_to_lower(node.op)
+
+        # Check if specialized handler exists.
+        if handler_name in dir(cls):
+          method_to_call = getattr(cls, handler_name)
+          ops_proto.append(method_to_call(node))
+
+    return helper.make_graph(ops_proto,
+                            "onnx_graph",
```
Comment (arpith-jacob) [inline onnx_tf/frontend.py]: Can you pass the name as an argument to this function?

File: onnx_tf/frontend.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  # DT_UINT64 = 23;
+  ]
+
+  @classmethod
+  def op_name_to_lower(cls, name):
+    return re.sub('(?<!^)(?=[A-Z])', '_', name).lower()
+
+  @classmethod
+  def tensorflow_graph_to_onnx_graph(cls, graph_def, output):
+    inputs_proto = []
```
Comment (arpith-jacob) [inline onnx_tf/frontend.py]: It's not clear to me what 'output' and 'inputs_proto' mean.

File: onnx_tf/frontend.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        if handler_name in dir(cls):
+          method_to_call = getattr(cls, handler_name)
+          ops_proto.append(method_to_call(node))
+
+    return helper.make_graph(ops_proto,
+                            "onnx_graph",
+                            inputs_proto,
+                            [helper.make_tensor_value_info(output.name,
+                                                          cls.onnx_types[output.attr["T"].type],
+                                                          [10])])
```
Comment (arpith-jacob) [inline onnx_tf/frontend.py]: Why the magic number [10]?

**Comment by tjingrant** [comment]
I'm working on reversing the trivial maps from frontend.
After that we can distribute work as before.

File: test/frontend/test_node.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+                    .astype(np.float32)
+
+  def test_relu(self):
+    shape = (10, 10)
+    x = tf.placeholder(tf.float32, shape=shape)
+    y = tf.nn.relu(x)
+
+    tf_graph = y.graph.as_graph_def(add_shapes=True)
+    for node in tf_graph.node:
+        if node.name == "Relu":
```
Comment (doru1004) [inline test/frontend/test_node.py]: Can we make this more robust or find a way of getting the output info directly from y?

**Pull request merged**